Source/Fuse.Controls.Navigation/NavigationControl.BlockInput.uno

```diff
@@ -0,0 +1,64 @@
+using Fuse.Triggers;
+
+namespace Fuse.Controls
+{
+	public enum NavigationControlBlockInput
+	{
+		/** Input is not blocked */
+		Never,
+		/** Input is blocked during navigation -- when `WhileNavigating` would be active */
+		WhileNavigating,
+	}
+	
+	public abstract partial class NavigationControl
+	{
+		NavigationControlBlockInput _blockInput = NavigationControlBlockInput.WhileNavigating;
+		/**
+			Whether or not input to block input to the pages during navigation.
+			
+			The default is `WhileNavigating`: the pages will not get any pointer input during navigation.
+		*/
+		public NavigationControlBlockInput BlockInput
+		{
+			get { return _blockInput; }
+			set
+			{
+				if (_blockInput == value)
+					return;
+					
+				_blockInput = value;
+				if (IsRootingCompleted)
+					UpdateBlockInput();
+			}
+		}
+
+		Trigger _blockInputTrigger;
+		void BlockInputRooted()
+		{
+			UpdateBlockInput();
+		}
+		
+		
+		void BlockInputUnrooted()
+		{
+			UpdateBlockInput(true);
+		}
+		
+		void UpdateBlockInput(bool forceOff = false)
+		{
+			if (forceOff || _blockInput == NavigationControlBlockInput.Never)
```

Comment:
I think I would separate this a bit, like so:

```Uno
void UpdateBlockInput()
{
	if (_blockInput == NavigationControlBlockInput.Never)
		DisableBlockInput();
	else
		EnableBlockInput();
}

void EnableBlockInput()
{
	if (_blockInputTrigger == null)
	{
		_blockInputTrigger = new NavigationInternal.BlockInputWhileNavigating(this);
		Children.Add(_blockInputTrigger);
	}
}

void DisableBlockInput()
{
	if (_blockInputTrigger != null)
	{
		Children.Remove(_blockInputTrigger);
		_blockInputTrigger = null;
	}
}
```

This gets us rid of the `forceOff`-argument, and we can just call `DisableBlockInput()` from `BlockInputUnrooted()` instead.

Reply:
Yeah, I'm really not sure why I used the flag.